src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -343,29 +296,40 @@ private void PushSelectedItemsToPresenter(NavigationBarSelectedTypeAndMember sel
                 listOfLeft,
                 newLeft,
                 newRight);
-            _versionStampOfFullListPushedToPresenter = null;
         }
 
         private void OnItemSelected(object? sender, NavigationBarItemSelectedEventArgs e)
         {
             AssertIsForeground();
+            _ = OnItemSelectedAsync(e.Item);
```

Comment:
switched to async/await to really clean up things.

Reply:
note that fire-and-forget is ok here as we do track cancellatino/failure happening in the task below.

Reply:
should this be tracked by an async operation listener as well?  Don't know if we have integration tests that select an item and need to wait for the processing to complete.